File: src/main/java/site/metacoding/weathertest/Weather.java
package site.metacoding.weathertest;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Weather {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String MSRDT; // 측정일시
    private String MSRRGN_NM; // 권역명
    private String MSRSTE_NM; // 측정소명
    private String PM10; // 미세먼지
    private String PM25; // 초미세먼지농도
    private String O3; // 오존
    private String NO2; // 이산화질소농도
    private String CO; // 일산화탄소농도
    private String SO2; // 아황산가스농도
    private String IDEX_NM; // 통합대기환경등급
    private String IDEX_MVL; // 통합대기환경지수
    private String ARPLT_MAIN; // 지수결정물질
}

File: src/main/weather.py
from pymysql import connect, cursors
import requests

# mariaDB 연결
conn = connect(host="localhost", user="green",
               password="green1234", db="greendb", charset="utf8")

cursor = conn.cursor()

# 공공데이터 담기
response = requests.get(
    "http://openapi.seoul.go.kr:8088/5875646654616b6935387676784375/json/RealtimeCityAir/1/25")


# 데이터 받기
responseParse = response.json()["RealtimeCityAir"]["row"]

# DB INSERT
insert_sql = "INSERT INTO weather(MSRDT, MSRRGN_NM, MSRSTE_NM, PM10, PM25, O3, NO2, CO, SO2, IDEX_NM, IDEX_MVL, ARPLT_MAIN) VALUES (%(MSRDT)s, %(MSRRGN_NM)s, %(MSRSTE_NM)s, %(PM10)s, %(PM25)s, %(O3)s, %(NO2)s, %(CO)s, %(SO2)s, %(IDEX_NM)s, %(IDEX_MVL)s, %(ARPLT_MAIN)s)"
cursor.executemany(insert_sql, responseParse)
conn.commit()
